app/src/androidTest/java/org/mozilla/fenix/ui/NoNetworkAccessStartupTests.kt

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui
+
+import androidx.core.net.toUri
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.IdlingRegistry
+import org.junit.After
+import org.junit.Test
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestHelper.packageName
+import org.mozilla.fenix.helpers.TestHelper.setNetworkEnabled
+import org.mozilla.fenix.helpers.TestHelper.verifyUrl
+import org.mozilla.fenix.helpers.idlingresource.NetworkConnectionIdlingResource
+import org.mozilla.fenix.ui.robots.browserScreen
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+/**
+ * Tests to verify some main UI flows with Network connection off
+ *
+ */
+
+class NoNetworkAccessStartupTests {
+    private val activityTestRule = HomeActivityTestRule()
+    private val networkDisconnectedIdlingResource = NetworkConnectionIdlingResource(false)
+    private val networkConnectedIdlingResource = NetworkConnectionIdlingResource(true)
+
+    @After
+    fun tearDown() {
+        IdlingRegistry.getInstance().unregister(networkDisconnectedIdlingResource)
+        // Restoring network connection and waiting to be back online
+        setNetworkEnabled(true)
+        IdlingRegistry.getInstance().register(networkConnectedIdlingResource)
+        Espresso.onIdle {
+            IdlingRegistry.getInstance().unregister(networkConnectedIdlingResource)
+        }
+
+        activityTestRule.finishActivity()
```

Comment:
nit: I'd think this happens automatically. The `finishActivity` gets called after `base.evaluate`, which I think is where the test content actually runs: https://cs.android.com/androidx/android-test/+/master:runner/rules/java/androidx/test/rule/ActivityTestRule.java;l=549;drc=bf4215f8816fc81db3d105ef2fbc4f748619ab45

So I think this can be removed

Reply:
done